java/daikon/chicory/Runtime.java

```diff
@@ -129,7 +130,9 @@ public CallInfo (int nonce, boolean captured) {
     }
 
     /** Stack of active methods. **/
-    private static Map<Thread,Stack<CallInfo>> thread_to_callstack
+    // /*@GuardedBy("Runtime.class")*/ Map<Thread, /*@GuardedBy("Runtime.class")*/ Stack</*@GuardedBy("Runtime.class")*/ CallInfo>> is the ideal annotated type,
+    // however this is not possible since Map is not a proper parameterized type.
```

Comment:
Sorry, this comment was misleading and I removed it. The issue is that the declaration of `Map` is `public interface Map<K extends Object, V extends Object>`.
Ideally we would be able to annotate it as
`public interface Map<K extends Object, V extends @GuardedByUnknown Object>` in order to allow Maps to store guarded values.
However, this would imply that Maps can only store and retrieve references to values but never dereference them, which is too restrictive.